addons/website_slides/static/src/js/public/components/slide_upload_dialog/slide_upload_dialog.xml

```diff
@@ -0,0 +1,397 @@
+                        >
+                            Create New Tag "<i t-out="select.data.searchValue"/>"
+                    </DropdownItem>
+                </t>
+            </SelectMenu>
+        </div>
+        <div class="mb-3">
+            <label for="duration" class="col-form-label">Estimated Completion Time</label>
+            <div class="input-group">
+                <input t-model="state.duration" type="number" id="duration" min="0" name="duration" placeholder='e.g. "15"' class="form-control" autocomplete="off"/>
+                <span class="input-group-text">Minutes</span>
+            </div>
+        </div>
+    </t>
+
+    <t t-name="website_slides.UploadSourceTypes">
+        <div class="mb-3">
+            <label for="source_type" class="col-form-label"><t t-out="source_type_label"/></label><br/>
+            <div class="form-check ms-2">
+                <input class="form-check-input" type="radio" name="source_type" id="source_type_local_file" data-value="local_file" checked="checked" t-on-click="() => this.onClickSourceType(true)"/>
+                <label class="form-check-label fw-normal" for="source_type_local_file">
+                    Upload from Device
+                </label>
+            </div>
+            <div class="form-check ms-2">
+                <input class="form-check-input" type="radio" name="source_type" id="source_type_external" data-value="external" t-on-click="() => this.onClickSourceType(false)"/>
+                <label class="form-check-label fw-normal" for="source_type_external">
+                    Retrieve from Google Drive
+                </label>
+            </div>
+        </div>
+        <t t-if="state.isLocalSource">
+            <div class="mb-3 o_wslides_js_slide_upload_local_file">
+                <label for="upload" class="col-form-label"><t t-out="select_file_label"/></label>
+                <input id="upload" name="file" class="form-control h-100" t-att-accept="accepted_files" type="file" required="required" t-on-change.prevent="onChangeFileInput"/>
+            </div> 
+        </t>
+        <t t-else="">
+            <div class="mb-3 o_wslides_js_slide_upload_external">
+                <label for="document_google_url" class="col-form-label"><t t-out="url_input_label"/></label>
+                <input t-model="state.url" t-att-id="url_input_name" t-att-name="url_input_name" class="form-control h-100" autocomplete="off"
+                    placeholder='e.g "https://drive.google.com/file/..."' required="required" t-on-change="onChangeUrl"/>
+            </div>
+        </t>
+    </t>
+
+    <!-- Slide Category templates -->
+
+    <t t-name="website_slides.UploadDialogDocument">
+        <div>
+            <form t-att-class="{'clearfix': true, 'was-validated': state.wasValidated}" t-on-submit.prevent="(ev) => ev.preventDefault()">
+                <div class="row">
+                    <div id="o_wslides_js_slide_upload_left_column" class="col-md-6">
```

Comment:
Those lines are common to all the upload templates, would be great to create a template for that

Reply:
So true! I'll keep that in mind and the other remarks on template refactoring while waiting for the overall refactoring discussions with @flch-odoo  :+1: